src/cli.ts

```diff
@@ -157,14 +158,23 @@ async function prepareSuites(inputs: string[]) {
     requireSuites(suites);
   }
 
+  /**
+   * Use the NODE_ENV variable to control the environment if its not explicity
+   * passed from either CLI or through the API
+   */
+  const environment = options.environment || process.env['NODE_ENV'];
+  /**
+   * Validate and handle configs
+   */
+  const config = readConfig(environment, options.config);
+  const params = config.params || JSON.parse(options.suiteParams);
   /**
    * use JSON reporter if json flag is enabled
    */
   const reporter = options.json ? 'json' : options.reporter;
 
   const results = await run({
-    params: JSON.parse(options.suiteParams),
-    environment: options.environment,
+    params: Object.freeze({ ...params, environment }),
```

Comment:
Glad to see we are freezing this object here that you prevent a lot of errors. So